tests/testcore/onebox.go

```diff
@@ -262,13 +226,12 @@ func newTemporal(t *testing.T, params *TemporalParams) *TemporalImpl {
 		hostsByProtocolByService:         params.HostsByProtocolByService,
 	}
 
-	for k, v := range staticOverrides {
+	for k, v := range dynamicConfigOverrides {
```

Comment:
It feels like it should be possible to merge the "global test overrides" and the ones from "params" (the per-suite ones?) earlier so that this code here is just one loop. Did you consider that?

Like maybe change https://github.com/temporalio/temporal/pull/7096/files#diff-b77e5f4402c434c0f9f4a58cbe72c7fa38e280e235f5d6c393b370301355cd3eR211 to `s.testClusterConfig.DynamicConfigOverrides = maps.Clone(dynamicConfigOverrides)`?